### PR TITLE
Fix clustering with DTW and ETImporter PP

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by wangcj05** [comment]
```
**** SUMMARY ****
*********************************
**** THE FOLLOWING TESTS CHANGED OUTCOMES ****
**** Test framework/PostProcessors/DataMiningPostProcessor/Clustering.agglomerativeDTW | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/DataMiningPostProcessor/Clustering.time_dep_hierarchicalScipy_DTW | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/DataMiningPostProcessor/Clustering.time_dep_hierarchicalScipy_DTW_derivative | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/ETimporterPostProcessor.ET_importer_OpenPSA | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/ETimporterPostProcessor.ET_importer_OpenPSA_Define_Branch | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/ETimporterPostProcessor.ET_importer_OpenPSA_Symbolic | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/ETimporterPostProcessor.ET_importer_OpenPSA_coupledET | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/TemporalDataMiningPostProcessor/Clustering.AffinityPropogation | has changed outcome from <FAILED> to <OK> !!!!
*****************
Number of success first file:
434
Number of success second file:
442
NUMBER OF SUCCESSES INCREASED: +8
TOTAL TIME OF TESTS DECREASED       : reference_time(756.2) > new_time(739.1)
AVERAGE TIMES FOR SUCCESS TESTS ARE : reference_time(1.74239631336) v.s. new_time(1.6721719457)
*********************************
```

**Comment by wangcj05** [comment]
This PR# is ready to review.

File: tests/framework/PostProcessors/TemporalDataMiningPostProcessor/Clustering/test_tdmSKLearnAffinityPropogation.xml [hunk 1]
```diff
@@ -145,7 +145,8 @@
 
   <DataObjects>
     <HistorySet name='clusterInfo'>
-      <Output>x1,x2,labels</Output>
```
Comment (mandd) [inline tests/framework/PostProcessors/TemporalDataMiningPostProcessor/Clustering/test_tdmSKLearnAffinityPropogation.xml]: is there a reason why the labels have been removed?

File: framework/PostProcessors/ETImporter.py [hunk 2]
```diff
@@ -209,7 +209,7 @@ def checkETstructure(self,links,listETs,connectivityMatrix):
     if len(zeroColumns)==0:
       self.raiseAnError(IOError, 'No root ET')
     if len(zeroColumns)==1:
-      rootETID = listETs[zeroColumns]
+      rootETID = listETs[zeroColumns[0]]
```
Comment (mandd) [inline framework/PostProcessors/ETImporter.py]: why this change?

Comment (wangcj05) [reply framework/PostProcessors/ETImporter.py]: The ```zeroColumns``` is a 1-D array, and we can not directly use it in the list.

File: tests/framework/PostProcessors/TemporalDataMiningPostProcessor/Clustering/test_tdmSKLearnAffinityPropogation.xml (hunk 1 above)
Comment (wangcj05) [reply tests/framework/PostProcessors/TemporalDataMiningPostProcessor/Clustering/test_tdmSKLearnAffinityPropogation.xml]: I moved the labels from Input to Output in my previous modifications, in this case it is consistent  with output HistorySet, we keep the labels in Output space. However, in the devel, we do keep the labels in the input space for the solutionExport, and in this modification, I moved the labels to the Input space.

**Comment by alfoa** [comment]
Approved...
